.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow.
+
+inputs:
+  install_dir:
+    description: 'The directory to install the binary.'
+    required: true
+  download_url:
+    description: 'The download URL for the binary to install.'
+    required: true
+  file_ext:
+    description: 'The type of file. Allowed values: tar, zip, tgz, none'
+    default: 'tgz'
+    required: false
+  enable_cache:
+    description: 'Enable the use of caching for the binary.'
+    default: 'false'
+    required: false
+  cache_key:
+    description: 'The URL of the binary to download.'
```

Comment:
Is this description right? Why are we asking for a key when we can generate one? Perhaps we should use a reasonable default key and allow user to override it?

Reply:
Copy paste issue, maybe we accept a name parameter and default to `${{ runner.os }}_${{ runner.arch }}_${{ inputs.name }}`?

Reply:
That sounds reasonable to me.